backend/controller/encryption/encryption.go

```diff
@@ -0,0 +1,72 @@
+package encryption
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/TBD54566975/ftl/backend/controller/encryption/dal"
+	"github.com/TBD54566975/ftl/backend/libdal"
+	"github.com/TBD54566975/ftl/internal/encryption"
+)
+
+type Service struct {
```

Comment:
It does feel like this should just be in `internal/encryption`, somehow decoupled from the concrete DAL implementation, but I think this is okay for now.

Reply:
Yeah I think so too. Maybe we just construct the dal, then use it to create and verify the `DataEncryptor`, then just pass around this interface